lib/lotus/validations/error.rb

```diff
@@ -63,21 +60,16 @@ class Error
 
       # Initialize a validation error
       #
-      # @param attribute_name [Symbol] the name of the attribute
       # @param validation [Symbol] the name of the validation
       # @param expected [Object] the expected value
       # @param actual [Object] the actual value
-      # @param namespace [String] the optional namespace
       #
       # @since 0.1.0
       # @api private
-      def initialize(attribute_name, validation, expected, actual, namespace = nil)
-        @attribute_name = attribute_name.to_s
+      def initialize(validation, expected, actual)
```

Comment:
@rodzyn More on this. In your [comment above](https://github.com/lotus/validations/pull/49#issuecomment-106631885) I see a snippet of code:

```ruby
validator.errors

=> #<Lotus::Validations::Errors:0x007fbb6485daf0
 @errors=
  {:first_name=>
    [#<Lotus::Validations::Error:0x007fbb6485d8c0 @actual="z", @expected="z", @validation=:exclusion>,
     #<Lotus::Validations::Error:0x007fbb6485d668 @actual="z", @expected="x", @validation=:inclusion>],
   :last_name=>[#<Lotus::Validations::Error:0x007fbb6485cf88 @actual=nil, @expected=true, @validation=:presence>]}>
```

When we do `errors.each` we yield a flatten array of `Error` instances. We loose that data structure (`Hash`), where the attribute name is the key for an array of errors (`:first_name => [...]`).

By removing attribute name, the `Error` instances that we yield, **they loose the reference with the attribute**. Look at the following error, how you can tell which is the attribute that caused it?

```ruby
#<Lotus::Validations::Error:0x007fbb6485d8c0 @actual="z", @expected="z", @validation=:exclusion>
```

Can you please restore it? Thank you.